src/Attachment/File.php

```diff
@@ -54,19 +60,21 @@ class File
      * @param string|null  $disk
      * @param string|null  $group
      */
-    public function __construct(UploadedFile $file, string $disk = null, string $group = null)
+    public function __construct(UploadedFile $file, string $disk = null, string $path = null, string $group = null)
```

Comment:
If we add this argument in the middle, then we break backward compatibility for users. 
Can we do it as a method?

```php
$upload = (new File($file, $this->disk))
    ->path('custom/path/etc')
    ->load();
```

Reply:
Yes , you have a valid point ,

i can add the path method , but 
- the generator construct has to change anyway ( $path param at the end )
- I will have to create a new generator object in the path method since its first instanciated in the constructor
### __construct() in Generator

```php
public function __construct(UploadedFile $file, string $path = null)
```
### path() in File
```php
/**
     * set a custom Path
     * @return File
     */
    public function path(String $path) {
        $generator = config('platform.attachment.generator', Generator::class);
        $this->engine = new $generator($this->file, $path);
        return $this;
    }
```


to prevent re-instantiating the generator we can add a setter for the path variable within  the Generator class
and then we can do something like this

```php
public function path(String $path) {
                $this->engine->setPath($path)
}
```